tests/plugins/dialogadvtab/dialog.js

```diff
@@ -68,7 +76,64 @@
 
 				dialog.hide();
 			} );
+		},
+
+		// (#1046).
+		'test adding two subsequent id changes': function() {
+			var bot = this.editorBot;
+
+			this.editorBot.setHtmlWithSelection( '<p>[aaa] <strong>bbb</strong></p>' );
+
+			// First add id to one of links.
+			bot.dialog( 'link', function( dialog ) {
+				var editor = this.editor;
+
+				dialog.setValueOf( 'info', 'url', 'a' );
+				dialog.setValueOf( 'advanced', 'advId', 'aa' );
+				dialog.getButton( 'ok' ).click();
+
+				assert.beautified.html( '<p><a href="http://a" id="aa">aaa</a> <strong>bbb</strong></p>',
+					bot.getData( true ) );
+
+				editor.getSelection().selectElement( editor.editable().findOne( 'strong' ) );
+
+				// Now add the id to a second link.
+				bot.dialog( 'link', function( dialog ) {
+					dialog.setValueOf( 'info', 'url', 'b' );
+					dialog.setValueOf( 'advanced', 'advId', 'bb' );
+					dialog.getButton( 'ok' ).click();
```

Comment:
It would be nice to extract duplicated linking code into a separate function.

Reply:
I have checked how it looks like, you can see it with 99b1c50a2c - that's on [t/1046b](https://github.com/ckeditor/ckeditor-dev/tree/t/1046b) branch.

TBH I find it to be less readable than the initial code - thus personally I don't think it simplifies the code.